index.js

```diff
@@ -43,42 +47,56 @@ module.exports = ${existingContents.trimRight()};\n`;
   fs.writeFileSync(newConfigPath, updatedContents, { encoding: 'utf-8' });
 
   fs.unlinkSync(oldConfigPath);
-}
 
-async function updateOptionalFeatures(root) {
-  let packageJSONPath = path.join(root, 'package.json');
-  let pkg = JSON.parse(fs.readFileSync(packageJSONPath, { encoding: 'utf-8' }));
-
-  let configPath = 'config/optional-features.json';
-  if (pkg['ember-addon'] && pkg['ember-addon'].configPath) {
-    configPath = pkg['ember-addon'].configPath;
-  }
+  console.log(
+    chalk.green(
+      `Migrated ${chalk.underline('.ember-cli')} to ${chalk.underline('.ember-cli.js')}\n`
+    )
+  );
+}
 
-  let config = {};
-  if (fs.existsSync(configPath)) {
-    let configContents = fs.readFileSync(configPath, { encoding: 'utf-8' });
-    config = JSON.parse(configContents);
+async function updateOptionalFeatures() {
+  console.log(
+    `About to update ${chalk.underline(
+      'config/optional-features.json'
+    )} you may be prompted for codemods`
+  );
+  const features = [
+    ['default-async-observers', true],
+    ['jquery-integration', false],
+    ['template-only-glimmer-components', true],
+    ['application-template-wrapper', false],
+  ];
+
+  for (let i = 0; i < features.length; i++) {
+    const feature = features[i];
+    await emberFeature(...feature);
   }
+  console.log();
+}
 
-  config['application-template-wrapper'] = false;
-  config['default-async-observers'] = true;
-  config['jquery-integration'] = false;
-  config['template-only-glimmer-components'] = true;
-
-  let keys = Object.keys(config);
-  let newConfig = {};
-  keys.sort().forEach(key => (newConfig[key] = config[key]));
-
-  fs.writeFileSync(configPath, JSON.stringify(newConfig, null, 2), { encoding: 'utf-8' });
+async function emberFeature(feature, enable) {
+  return new Promise((resolve, reject) => {
+    const enableString = `${enable ? 'en' : 'dis'}able`;
```

Comment:
I think this is _probably_ easier to parse (and ironically, is shorter):

```suggestion
    const enableString = enable ? 'enable' : 'disable';
```